bin/po2android.js

```diff
@@ -63,79 +76,120 @@ const getUniqueName = ( function() {
+			if ( comment ) {
+				localizedEntry += `${ indent }<!-- ${ comment.replace(
+					'--',
+					'—'
+				) } -->\n`;
+			}
+			if ( stringPlural ) {
+				localizedEntry += `${ indent }<string-array name="${ uniqueName }" tools:ignore="UnusedResources">
 ${ indent }${ indent }<item>${ escapedValue }</item>
 ${ indent }${ indent }<item>${ escapedValuePlural }</item>
 ${ indent }</string-array>
 `;
-		} else {
-			localizedEntry += `${ indent }<string name="${ uniqueName }" tools:ignore="UnusedResources">${ escapedValue }</string>\n`;
-		}
-		result[ uniqueName ] = localizedEntry;
-		return result;
-	}, {} );
+			} else {
+				localizedEntry += `${ indent }<string name="${ uniqueName }" tools:ignore="UnusedResources">${ escapedValue }</string>\n`;
+			}
+			result[ uniqueName ] = localizedEntry;
+			return result;
+		},
+		{}
+	);
 	// try to minimize changes in diffs by sorting strings
 	const androidResourcesSortedList = Object.entries( androidResourcesMap )
 		.sort( ( left, right ) => left[ 0 ].localeCompare( right[ 0 ] ) )
 		.map( ( entry ) => entry[ 1 ] );
-	return `<?xml version="1.0" encoding="utf-8"?>\n<resources xmlns:tools="http://schemas.android.com/tools">\n${ androidResourcesSortedList.join( '' ) }</resources>\n`;
+	return `<?xml version="1.0" encoding="utf-8"?>\n<resources xmlns:tools="http://schemas.android.com/tools">\n${ androidResourcesSortedList.join(
+		''
+	) }</resources>\n`;
 }
 
+// Returns used strings of a specified domain
+const getUsedStrings = ( usedStringsFile ) => {
+	return require( path.resolve( usedStringsFile ) );
+};
+
 if ( require.main === module ) {
-	if ( process.stdin.isTTY ) {
-		const potFileName = process.argv[ 2 ];
-		const destination = process.argv[ 3 ];
-		const potFileContent = fs.readFileSync( potFileName );
-		const xmlOutput = po2Android( potFileContent, process.argv[ 3 ] );
-		fs.writeFileSync( destination, xmlOutput );
-	} else {
-		let inputData = '';
-		process.stdin.on( 'readable', function() {
-			const chunk = this.read();
-			if ( chunk !== null ) {
-				inputData += chunk;
-			}
-		} );
-		process.stdin.on( 'end', function() {
-			process.stdout.write( po2Android( inputData ) );
-		} );
+	const destination = process.argv[ 2 ];
+	const usedStringsFile = process.argv[ 3 ];
+
+	const usedStrings = getUsedStrings( usedStringsFile );
+	let flattenUsedStrings = Object.entries( usedStrings ).reduce(
+		( result, [ domain, strings ] ) => {
+			return [ ...result, ...Object.values( strings ) ];
+		},
+		[]
+	);
+	flattenUsedStrings = [ ...new Set( flattenUsedStrings ) ];
+
+	const nativeStringsWithContext = [];
+	let onlyNativeStrings = flattenUsedStrings.filter( ( item ) => {
+		const { platforms, string } = item;
+
+		const isNative =
+			! platforms.includes( 'web' ) && platforms.includes( 'android' );
+		const hasContext = !! item.context;
+		if ( isNative && hasContext ) {
+			nativeStringsWithContext.push( string );
+		}
+
+		return isNative && ! hasContext;
+	}, [] );
```

Comment:
We only add the localization strings files of strings referenced in `.native.js` and `.android.js files`, and that doesn't have context.

Reply:
The current list of strings that we won't be able to translate are:
```
[
  'Link rel attribute value placeholder\x04None',
  'inline textinput cell\x04%s. Empty',
  'inline textinput cell\x04%1$s, %2$s. Empty',
  'inline textinput cell\x04%1$s, %2$s',
  'inline textinput cell\x04%1$s, %2$s, %3$s',
  'switch control\x04%s. On',
  'switch control\x04%s. Off',
  'switch control\x04%1$s, %2$s. On',
  'switch control\x04%1$s, %2$s. Off'
]
```

**NOTE:** The strings contain both the value and context, it has the following format: `<CONTEXT>\x04<STRING>`.